components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/ContextLoader.java

```diff
@@ -57,11 +54,7 @@ public class ContextLoader {
      */
     public static String getTenantDomainFromContext() {
 
-        String tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
-        if (IdentityUtil.threadLocalProperties.get().get(TENANT_NAME_FROM_CONTEXT) != null) {
-            tenantDomain = (String) IdentityUtil.threadLocalProperties.get().get(TENANT_NAME_FROM_CONTEXT);
-        }
-        return tenantDomain;
+        return IdentityTenantUtil.resolveTenantDomain();
```

Comment:
instead of changing the behavior of the public method, shall we refactor its usages in the repo and use the new IdentityTenantUtil method?

Reply:
Addressed with commit 2aa6620b6dade10276593710a6f25d744edf34b6